### ISSUE TITLE
Notion integration as a data source not working for SELF_HOSTED version

### ISSUE THREAD

**Issue by AShabana** [langgenius/dify#641]
Dify version: Self Host

## Steps To Reproduce

1. Clone / docker compose up
2. Create dateset type "Sync From Notion"

## The current behavior
<img width="1008" alt="Screenshot 2023-07-25 at 9 25 26 AM" src="https://github.com/langgenius/dify/assets/7352089/09d51edb-c0dd-4b26-adb5-0465a76fff23">

## The expected behavior
<img width="1031" alt="Screenshot 2023-07-25 at 9 33 24 AM" src="https://github.com/langgenius/dify/assets/7352089/59864916-1b45-4a00-bef4-4c4e016c1af1">

**Comment by UpwardGrowth** [comment]
I encountered the same problem as you. I used the variables NOTION_CLIENT_ID and NOTION_CLIENT_SECRET to solve the issue of losing ClientId, but I still cannot redirect to the selection page of Notion. I hope someone can help us resolve our doubts.

**Comment by UpwardGrowth** [comment]
> I encountered the same problem as you. I used the variables NOTION_CLIENT_ID and NOTION_CLIENT_SECRET to solve the issue of losing ClientId, but I still cannot redirect to the selection page of Notion. I hope someone can help us resolve our doubts.

I can't jump to the page for selecting a page. Once I click on binding, it will redirect to http://127.0.0.1:3000/apps.

**Comment by AShabana** [comment]
I tested with redirect url as anything under localhost all failed with this message

![Screenshot 2023-07-25 at 11 23 47 AM](https://github.com/langgenius/dify/assets/7352089/696024ef-a926-4480-be37-903a5912c229)

**Comment by JohnJyong** [comment]
pls follow the below doc
![image](https://github.com/langgenius/dify/assets/76649700/9987bb2a-4cc2-4e1f-a26d-feb504b0b441)
https://docs.dify.ai/getting-started/faq/install-faq

**Comment by UpwardGrowth** [comment]
> > I encountered the same problem as you. I used the variables NOTION_CLIENT_ID and NOTION_CLIENT_SECRET to solve the issue of losing ClientId, but I still cannot redirect to the selection page of Notion. I hope someone can help us resolve our doubts.
> 
> I can't jump to the page for selecting a page. Once I click on binding, it will redirect to http://127.0.0.1:3000/apps.



> pls follow the below doc ![image](https://user-images.githubusercontent.com/76649700/255852934-9987bb2a-4cc2-4e1f-a26d-feb504b0b441.png) https://docs.dify.ai/getting-started/faq/install-faq

Yes, this document is the correct answer. Since I am running locally, I solved the issue of missing ClientId by using the following two parameters:

NOTION_INTEGRATION_TYPE=internal
NOTION_INTERNAL_SECRET=your-secret

Regarding the solution to my other problem, even though I filled in the secret and restarted, it still couldn't retrieve data from Notion. Later, I found out that it was because I didn't configure the connections. You need to configure connections in your document to connect to the integrations you created.
